sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,114 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.process.Props;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public final class Cluster {
+  private static final Logger LOG = LoggerFactory.getLogger(Cluster.class);
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  private boolean active = false;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param props the props
+   */
+  protected Cluster(Props props) {
+    ClusterProperties.completeDefaults(props);
+
+    if (props.valueAsBoolean(ClusterProperties.ENABLED)) {
+      initializeHazelcast(props);
+    }
+  }
+
+  private void initializeHazelcast(Props props) {
+    Config hzConfig = new Config();
+    // Configure the network instance
+    NetworkConfig netConfig = hzConfig.getNetworkConfig();
+    netConfig.setPort(props.valueAsInt(ClusterProperties.PORT))
+      .setPortAutoIncrement(props.valueAsBoolean(ClusterProperties.PORT_AUTOINCREMENT));
+
+    if (props.value(ClusterProperties.INTERFACES) != "") {
+      netConfig.getInterfaces()
+        .setEnabled(true)
+        .setInterfaces(
+          Arrays.asList(props.value(ClusterProperties.INTERFACES).split(","))
+        );
+    }
+
+    // Only allowing TCP/IP configuration
+    JoinConfig joinConfig = netConfig.getJoin();
+    joinConfig.getAwsConfig().setEnabled(false);
+    joinConfig.getMulticastConfig().setEnabled(false);
+    joinConfig.getTcpIpConfig().setEnabled(true);
+
+    // Add the defined members of the cluster
+    if (props.value(ClusterProperties.MEMBERS) != "") {
+      for (String member : props.value(ClusterProperties.MEMBERS).split(",")) {
+        if (!member.contains(":")) {
+          // If member does not define his port, let's use our default one
+          member += ":" + ClusterProperties.DEFAULT_PORT;
+        }
+        joinConfig.getTcpIpConfig().addMember(member);
+      }
+    }
+
+    // Tweak HazelCast configuration
+    hzConfig
+      .setProperty("hazelcast.tcp.join.port.try.count", "10") // Increase the number of tries
+      .setProperty("hazelcast.socket.bind.any", "false") // Don't bind all interfaces
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)